src/language-server/project/base.ts

```diff
@@ -1,97 +1,51 @@
-import path, { extname } from "path";
-import { lstatSync, readFileSync } from "fs";
 import { URI } from "vscode-uri";
 
-import {
-  TypeSystemDefinitionNode,
-  isTypeSystemDefinitionNode,
-  TypeSystemExtensionNode,
-  isTypeSystemExtensionNode,
-  DefinitionNode,
-  GraphQLSchema,
-  Kind,
-} from "graphql";
+import { GraphQLSchema } from "graphql";
 
 import {
-  TextDocument,
   NotificationHandler,
   PublishDiagnosticsParams,
-  Position,
   CancellationToken,
-  CompletionItem,
-  Hover,
-  Definition,
-  ReferenceContext,
-  Location,
-  DocumentSymbol,
   SymbolInformation,
-  Range,
-  CodeAction,
-  CodeLens,
+  Connection,
+  ServerRequestHandler,
+  TextDocumentChangeEvent,
 } from "vscode-languageserver/node";
-
-import { GraphQLDocument, extractGraphQLDocuments } from "../document";
+import { TextDocument } from "vscode-languageserver-textdocument";
 
 import type { LoadingHandler } from "../loadingHandler";
 import { FileSet } from "../fileSet";
-import {
-  ApolloConfig,
-  ClientConfig,
-  isClientConfig,
-  isLocalServiceConfig,
-  keyEnvVar,
-  RoverConfig,
-} from "../config";
-import {
-  schemaProviderFromConfig,
-  GraphQLSchemaProvider,
-  SchemaResolveConfig,
-} from "../providers/schema";
-import { ApolloEngineClient, ClientIdentity } from "../engine";
+import { ApolloConfig, ClientConfig, RoverConfig } from "../config";
 import type { ProjectStats } from "../../messages";
 
 export type DocumentUri = string;
 
-const fileAssociations: { [extension: string]: string } = {
-  ".graphql": "graphql",
-  ".gql": "graphql",
-  ".js": "javascript",
-  ".ts": "typescript",
-  ".jsx": "javascriptreact",
-  ".tsx": "typescriptreact",
-  ".vue": "vue",
-  ".svelte": "svelte",
-  ".py": "python",
-  ".rb": "ruby",
-  ".dart": "dart",
-  ".re": "reason",
-  ".ex": "elixir",
-  ".exs": "elixir",
-};
-
-interface GraphQLProjectConfig {
-  clientIdentity: ClientIdentity;
+export interface GraphQLProjectConfig {
   config: ClientConfig | RoverConfig;
   configFolderURI: URI;
   loadingHandler: LoadingHandler;
 }
 
-export abstract class GraphQLProject implements GraphQLSchemaProvider {
-  public schemaProvider: GraphQLSchemaProvider;
+type ConnectionHandler = {
+  [K in keyof Connection as K extends `on${string}`
+    ? K
+    : never]: Connection[K] extends (
+    params: ServerRequestHandler<any, any, any, any> & infer P,
+    token: CancellationToken,
+  ) => any
+    ? P
+    : never;
+};
+
+export abstract class GraphQLProject {
```

Comment:
This class was split up into this `base.ts` and `internal.ts` - those "internal" changes could also have moved into the `ClientProject`, but that class is big enough as it is right now.